Main.py

```diff
@@ -1,82 +1,102 @@
+                        mutateORF = np.random.random([sum(toMutate), 1]) * (data1_Cum[-1]) # pick mutation
                         idxORF = np.searchsorted(data1_Cum, mutateORF, side='right') # peek mutation
                         mutation[toBirth[toMutate]] = idxORF  # store mutation
-                        #idxDivide = np.nonzero(toDivide)[0]
-                        #nr_mutations +=
+                        # idxDivide = np.nonzero(toDivide)[0]
+                        # nr_mutations +=
                         print 'time is ', t
                         print 'alive: ', nr_alive
-                        print data4[idxORF,iEnv]
-
+                        print data4[idxORF, iEnv]
                         simEnv['cell_cycle_time'][toBirth[toMutate]] = simEnv['cell_cycle_time'][idxDivide[toMutate]] + \
-                                                                       data4[idxORF,iEnv]*MEAN_CELL_CYCLE_TIME #(CELL_CYCLE_TIMES[simEnv['founder_id'][idxDivide[toMutate]]])
+                                                                       data4[idxORF, iEnv]*MEAN_CELL_CYCLE_TIME #(CELL_CYCLE_TIMES[simEnv['founder_id'][idxDivide[toMutate]]])
                         simEnv['next_divition'][toBirth[toMutate]] = t + simEnv['cell_cycle_time'][toBirth[toMutate]]
 
                 ## Population ages
                 simEnv['age'][0:nr_alive] += 0.1 # worry, might take up alotta time.
                 nr_alive += nr_Divide
-                #print 'alive: ', nr_alive
+                # print 'alive: ', nr_alive
 
                 # Check cycle exit condition
-                if nr_alive >= MAXIMUM_NR_AGENTS:
-                    stillInCycle = 0
-
-
-
-            # \todo: Sample for next cycle
+                if __name__ == '__main__':
+                    if __name__ == '__main__':
+                        if nr_alive >= MAXIMUM_NR_AGENTS-10:
+                            stillInCycle = 0
+        
+                # \todo: Save data for Growth curve
+                    # if iCycle == 0 or 24 or 49:
+                    # nr_alive at increments of 20 min. -> 30 points
+                    
+                # \todo: Save data for Evolutionary tree, Look plot code online. Binary tree!
+                    # mutations and mothers for all mutated cells
+                    # nr unmutated cells, 
+                    # time of mutation                        
+    
+                # \todo: Save data for histograms of GT
+                    # if iCycle .. first middle last
+                    # save hist GT
+                
+                # \todo: Save meanGT at end of cycles.
+                            
+            # Sample for next cycle
+            sample = np.random.random(SAMPLE_COUNT)*MAXIMUM_NR_AGENTS
+            simEnv[0:SAMPLE_COUNT] = simEnv[sample]
+            nr_alive = SAMPLE_COUNT
+            simEnv['next_divition'][0:nr_alive] = simEnv['cell_cycle_time'][0:nr_alive]
```

Comment:
> It's because I'm resetting the time at the end of the cycle, this would make calculations more accurate since we're dealing with float16.
> But we certainly can leave the time be if that's what you want, and yes then we would do as you suggest.

Your point about accuracy is correct by all means. It might be better to run this on `float32` to not have to do this to save compute time.

I considered `next_division` to be in experimental time e.g. after 2142.41 minutes of experiment this cell should have its next division. The shifting should in your implementation be:

`simEnv['next_divition'][0:nr_alive] -= MEASUREMENT_FREQUENCY`

I.e. I still don't see the logic of the line 97 above.

Plus then your `to_divide` would simply be:

`np.less_equal(simEnv['next_divition'][0:nr_alive], 0, out=to_divide)`

or 

`to_divide[...] = simEnv['next_divition'][0:nr_alive] <= 0`

For some reason they take the same time.

If you just assign it you will create a new object and this might be fine here but we want to make sure as far as possible that we keep using the same objects over and over.